app/src/main/java/com/kodeco/android/countryinfo/util/CountryInfoState.kt

```diff
@@ -0,0 +1,17 @@
+package com.kodeco.android.countryinfo.util
+
+import android.os.Parcelable
+import com.kodeco.android.countryinfo.model.Country
+import kotlinx.parcelize.Parcelize
+
+@Parcelize
+sealed class CountryInfoState : Parcelable {
+    class Success(val countries: List<Country>):CountryInfoState(){
+    }
+    class Error(val message: String?) : CountryInfoState(){
+    }
+    class CountryDetails(val country: Country, val countries: List<Country>):CountryInfoState(){
```

Comment:
Why are you passing a list of countries to the CountryDetails? If this is what the name suggests, then it's meant to hold data for a single country, not all countries. Plus, you are already passing a list of countries elsewhere.

Reply:
On the press of back button, i wanted to avoid making the api call again.